src/Program.cs

```diff
@@ -103,6 +105,43 @@ public static void RealTimeProcessingByProcessId(int pid, Options options, Confi
             }
         }
 
+        private static void SetupHeapStatsTimerIfEnabled(Configuration.Configuration configuration)
+        {
+            if (configuration.StatsMode == null)
+            {
+                return;
+            }
+
+            // Setup timer for Heap Stats.
+            // At this point of time, all validations are successful => get exactly what we need without checking.
+            string timerAsString = configuration.StatsMode["timer"];
+            int period = int.Parse(timerAsString[0..^1]);
+            char periodType = timerAsString.Last();
+            period = periodType switch
+            {
+                // minutes.
+                'm' => period * 60 * 1000,
+                // seconds.
+                's' => period * 1000,
+                // default, shouldn't get here as the validation takes care of this.
+                _ => throw new NotImplementedException()
+            };
+
+            TimerCallback timerCallback = (_) =>
+            {
+                if (lastGC != null)
+                {
+                    PrintLastStats();
+                }
+            };
+
+            // TODO: Figure out when to dispose this.
+            heapStatsTimer = new Timer(callback: timerCallback, 
```

Comment:
I don't think you need to dispose it - if you are using the period stats mode you'd need this timer as long as the process is running anyway.